src/main/java/com/yellow/ordermanageryellow/service/TopProductService.java

```diff
@@ -0,0 +1,107 @@
+package com.yellow.ordermanageryellow.service;
+
+import com.mongodb.client.AggregateIterable;
+import com.yellow.ordermanageryellow.Dto.ProductAmountDto;
+import com.yellow.ordermanageryellow.Dto.TopProductDTO;
+import com.yellow.ordermanageryellow.Exception.NoDataException;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+import org.springframework.stereotype.Service;
+
+import org.bson.Document;
+
+@Service
+public class TopProductService {
+
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    public AggregateIterable<Document> aggregationTopSoldProduct() {
+        LocalDate currentDate = LocalDate.now();
+        LocalDate beginningOfCurrentMonth = currentDate.withDayOfMonth(1);
+        LocalDate threeMonthsAgo = beginningOfCurrentMonth.minusMonths(3);
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        String startDate = threeMonthsAgo.format(formatter);
+        String endDate = beginningOfCurrentMonth.format(formatter);
+        System.out.println(threeMonthsAgo);
+        AggregateIterable<Document> result = mongoTemplate.getCollection("Orders").aggregate(Arrays.asList(
+                new Document("$match",
+                        new Document("audit_data.create_date",
+                                new Document("$gte", startDate)
+                                        .append("$lt", endDate))
+                                .append("order_status", "DONE")),
```

Comment:
Please dont use "DONE"
use enum of order status

Reply:
I don't know how because the enum is not saved in Mongo as a name and the aggregations work directly

Reply:
Did you try it?
It should be work since birth of them are same string